src/codegen/events/solana.rs

```diff
@@ -0,0 +1,63 @@
+use crate::codegen::cfg::{ControlFlowGraph, Instr};
+use crate::codegen::encoding::create_encoder;
+use crate::codegen::encoding::AbiEncoding;
+use crate::codegen::events::EventEmitter;
+use crate::codegen::expression::expression;
+use crate::codegen::vartable::Vartable;
+use crate::codegen::{Expression, Options};
+use crate::sema::ast;
+use crate::sema::ast::{Function, Namespace, Type};
+use sha2::{Digest, Sha256};
+use solang_parser::pt::Loc;
+
+/// This struct implements the trait 'EventEmitter' to handle the emission of events for Solana.
+pub(super) struct SolanaEventEmitter<'a> {
+    pub(super) loc: Loc,
+    /// Arguments passed to the event
+    pub(super) args: &'a [ast::Expression],
+    pub(super) ns: &'a Namespace,
+    pub(super) event_no: usize,
+}
+
+impl EventEmitter for SolanaEventEmitter<'_> {
+    fn emit(
+        &self,
+        contract_no: usize,
+        func: &Function,
+        cfg: &mut ControlFlowGraph,
+        vartab: &mut Vartable,
+        opt: &Options,
+    ) {
+        let discriminator_image = format!("event:{}", self.ns.events[self.event_no].name);
```

Comment:
In the tests you have a dedicated function for this. Instead of having that code duplicated, it could be refactored into this module and then use it for the tests as well?

Reply:
If I add here and make it available for test, it would need to be `pub` so that we can access it anywhere. This function should be private to this module.

Reply:
Maybe it's a use case for `pub(crate)` visibility then?